File: app/src/main/java/ru/example/newsreader/room/Dao/ArticleDao.java
package ru.example.newsreader.room.Dao;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import java.util.List;

import ru.example.newsreader.room.Entity.ArticleEntity;

@Dao
public interface ArticleDao {
    @Insert
    void insert(ArticleEntity article);

    @Update
    void update(ArticleEntity article);

    @Delete
    void delete(ArticleEntity article);

    @Query("SELECT * FROM ArticleEntity")
    List<ArticleEntity> getAllArticles();

    @Query("DELETE FROM ArticleEntity")
    void deleteAll();

   /* @Query("SELECT * FROM ArticleEntity WHERE source=:source")
    List<ArticleEntity> findArticlesForSource(final String source);*/
}

File: app/src/main/java/ru/example/newsreader/room/dao/ArticleDao.kt
package ru.example.newsreader.room.dao

import android.arch.persistence.room.*
import ru.example.newsreader.room.entity.ArticleEntity

@Dao
interface ArticleDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(article : ArticleEntity)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAll(articleList: List<ArticleEntity>)

    @Update
    fun update(article: ArticleEntity)

    @Delete
    fun delete(article: ArticleEntity)

    @Query("SELECT * FROM ArticleEntity")
    fun getAllArticles() : List<ArticleEntity>

    @Query("DELETE FROM ArticleEntity")
    fun deleteAll()
}